### PR TITLE
feat(recording): frontend logic can support live streaming and recording

### PR DESCRIPTION
**Pull request by virtuacoplenny**
Instead of either live streaming or recording, now both can live together. The
changes to facilitate such include the following:
- Killing the state storing in Recording.js. Instead state is stored in the lib
  and updated in redux for labels to display the necessary state updates.
- Creating a new container, Labels, for recording labels. Previously labels were
  manually created and positioned. The container can create a reasonable number
  of labels and only the container itself needs to be positioned with CSS. The
  VideoQualityLabel has been shoved into the container as well because it moves
  along with the recording labels.
- The action for updating recording state has been modified to enable updating
  an array of recording sessions to support having multiple sessions.
- Confirmation dialogs for stopping and starting a file recording session have
  been created, as they previously were jquery modals opened by Recording.js.
- Toolbox.web displays live streaming and recording buttons based on
  configuration instead of recording availability.
- VideoQualityLabel and RecordingLabel have been simplified to remove any
  positioning logic, as the Labels container handles such.
- Previous recording state update logic has been moved into the RecordingLabel
  component. Each RecordingLabel is in charge of displaying state for a
  recording session. The display UX has been left alone.
- Sipgw availability is no longer broadcast so remove logic depending on its
  state. Some moving around of code was necessary to get around linting errors
  about the existing code being too deeply nested (even though I didn't touch
  it).

### CODE REVIEW

**Comment by virtuacoplenny** [comment]
@zbettenbuk, this definitely needs Brian's review/approval as it exercises his new Jibri logic. However, the recording related changes will likely affect your future work so I've added you as a reviewer so you know what mess you're going to get into.

**Comment by virtuacoplenny** [comment]
Relies on https://github.com/jitsi/lib-jitsi-meet/pull/741

**Comment by virtuacoplenny** [comment]
Force pushed to resolve conflicts in react/features/toolbox/components/Toolbox.web.js.

File: modules/UI/UI.js [hunk 1]
```diff
@@ -337,7 +337,7 @@ UI.start = function() {
     if (!interfaceConfig.filmStripOnly) {
         VideoLayout.initLargeVideo();
     }
-    VideoLayout.resizeVideoArea(true, true);
+    VideoLayout.resizeVideoArea(true, false);
```
Comment (bbaldino) [inline modules/UI/UI.js]: probably good to add a comment here

File: react/features/recording/actions.js [hunk 2]
```diff
@@ -1,66 +1,24 @@
...
+ *     type: RECORDING_SESSION_UPDATED,
+ *     session: Object
  * }}
  */
-export function updateRecordingState(recordingState = {}) {
+export function updateRecordingSession(session) {
     return {
-        type: RECORDING_STATE_UPDATED,
-        recordingState
+        type: RECORDING_SESSION_UPDATED,
```
Comment (bbaldino) [inline react/features/recording/actions.js]: i wonder if it's worth a comment talking about why we only fire certain members of the session object.  which also makes me think about how doing this implies certain things from the session should not be exposed...but by what metric would we determine what should and should not?  (thinking about someone in the future adding something, how would they know whether or not it should be exposed in the event?)

**Comment by virtuacoplenny** [comment]
Jenkins test this please

**Pull request merged**